integrations/wagmi-v2/src/index.ts

```diff
@@ -0,0 +1,93 @@
+import {
+	type SapphireAnnex,
+	wrap,
+	wrapEIP1193Provider,
+} from "@oasisprotocol/sapphire-paratime";
+import type { Transport } from "@wagmi/core";
+import { custom, injected } from "@wagmi/core";
+import type { EIP1193Provider } from "viem";
+
+type Window = {
+	ethereum?: EIP1193Provider;
+};
+
+const cachedProviders: Map<EIP1193Provider, EIP1193Provider> = new Map();
+
+/**
+ * Wrap the `window.ethereum` with the Sapphire encryption layer.
+ * Used to provide encrypted transactions and calldata to Wagmi browser dApps.
+ *
+ * Example:
+ * ```
+ *
+ *    import { injectedWithSapphire } from '@oasisprotocol/sapphire-wagmi';
+ *
+ *    export const config = createConfig({
+ *      connectors: [
+ *        injectedWithSapphire()
+ *      ],
+ *      ...
+ *    });
+ *
+ * ```
+ *
+ * @returns Same as injected()
+ */
+export function injectedWithSapphire(): ReturnType<typeof injected> {
+	return injected({
+		target: () => {
+			return {
+				id: "injected-sapphire",
+				name: "Injected (Sapphire)",
+				provider(window?: Window) {
```

Comment:
Is this window parameter necessary? Is it the expected function signature?

Reply:
Yes, it is expected:

https://github.com/wevm/wagmi/blob/8e9ee793c4f378de523239bb8db63cbf09cc980e/packages/core/src/connectors/injected.ts#L472